include/wex/presets/GameObject.hpp

```diff
@@ -0,0 +1,42 @@
+#include "../Common.hpp"
+#include "../Engine.hpp"
+#include "../util.hpp"
+#include "Component.hpp"
+#include <cstddef>
+#include <vector>
+
+namespace wex {
+
+/// \brief The game game object class. A game object is a bag of components with an
+/// `onUpdate` method that is called once per iteration of the game loop.
+class GameObject {
+	std::vector<std::unique_ptr<Component>> mComponents;
+
+  public:
+	GameObject() = default;
+
+	/// \brief called on every 'update' call to the parent engine
+	virtual void onUpdate([[maybe_unused]] double dt) {}
+
+	template <typename T>
+	T* get() {
```

Comment:
Should be a `const` member function (don't need non const overload, because pointer isn't const)

Reply:
Aaah, great catch!